src/box/lua/key_def.c

```diff
@@ -85,6 +85,11 @@ luaT_push_key_def_parts(struct lua_State *L, const struct key_def *key_def)
 		lua_pushnumber(L, part->fieldno + TUPLE_INDEX_BASE);
 		lua_setfield(L, -2, "fieldno");
 
+		assert(part->sort_order == SORT_ORDER_ASC ||
+		       part->sort_order == SORT_ORDER_DESC);
+		lua_pushstring(L, sort_order_strs[part->sort_order]);
+		lua_setfield(L, -2, "sort_order");
```

Comment:
The sort order shouldn't be reported as "asc" for a hash index.

```
tarantool> _ = box.schema.create_space('test')
---
...

tarantool> box.space.test:create_index('pk', {type = 'hash'})
---
- unique: true
  parts:
  - type: unsigned
    sort_order: asc
    is_nullable: false
    fieldno: 1
  id: 0
  space_id: 512
  type: HASH
  name: pk
...
```

Reply:
Done.